docs/reference/api/usb/USBHID.md

```diff
@@ -0,0 +1,56 @@
+## USBHID
+
+<span class="images">![](https://os.mbed.com/docs/v5.9/feature-hal-spec-usb-device-doxy/class_u_s_b_h_i_d.png)<span>USBHID class hierarchy</span></span>
+
+You can use the USBHID class to send and receive messages over USB. For example, you can define your own protocol and communicate between your computer and Mbed with all capabilities of USB communication. To use USBHID, you need a script running on the host side (computer). For example, on a 32-bit Windows 7 machine, you can use [pywinusb](https://github.com/rene-aguirre/pywinusb).
```

Comment:
What are some best practices for using USBHID? What are some concepts you need to understand to use it? What type of script do you need and why? Also, what are all the capabilities of USB communication? Do you not get them with other USB classes? Please elaborate.